include/deal.II/numerics/vector_tools.h

```diff
@@ -763,22 +763,25 @@ namespace VectorTools
    * values are disabled. The ordering of arguments to this function is such
    * that you need not give a second quadrature formula if you don't want to
    * project to the boundary first, but that you must if you want to do so.
+
+   * If the FiniteElement is supported by the MatrixFree class, neither
+   * @p enforce_zero_boundary nor @p project_to_boundary_first are requested,
+   * the FiniteElement has less than five components and its degree is less
+   * than nine, a MatrixFree implementation is used.
+   * Otherwise, only serial Triangulations are supported and this function needs
+   * the mass matrix of the finite element space on the present grid.
+   * To this end, the mass matrix is assembled exactly using
+   * MatrixTools::create_mass_matrix.
    *
-   * This function needs the mass matrix of the finite element space on the
-   * present grid. To this end, the mass matrix is assembled exactly using
-   * MatrixTools::create_mass_matrix. This function performs numerical
-   * quadrature using the given quadrature rule; you should therefore make
-   * sure that the given quadrature formula is also sufficient for the
-   * integration of the mass matrix.
+   * This function performs numerical quadrature using the given quadrature rule;
```

Comment:
is it still the case? I though you integrate mass matrix with `fe_degree+2` whereas the RHS can be done using user-provided quadrature formula.